### PR TITLE
Storybook - React components Styleguide - React tree component

### PR DESCRIPTION
**Pull request by lneves12**
## What does this PR change?

The first version of the react components styleguide! 

**Steps to try this out:**
- Adapt '.storybook/preview-head.html'
- `yarn run storybook` on web/html/src

This is still not self-sufficient because the styles need to be pointing to a VM (preview-head.html), but let's start iterating from here

**This also adds a Tree component and its stories implemented by @cbosdo in https://github.com/uyuni-project/uyuni/pull/684** 

**Future work:** Add a step to webpack to inline all the external resources in the preview-head.html. Having this we can use the storybook locally without a server dependency or even build a production static styleguide to deploy anywhere :)

## GUI diff

Screenshot from 2019-08-26 13-21-47
Screenshot from 2019-08-26 13-21-38

- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- [doc-susemanager](https://github.com/SUSE/doc-susemanager) PR or issue was created (GitHub automatic link expected below)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint" (Test skipped, there are no changes to test)		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/.storybook/preview-head.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<script src="https://head-server.tf.local/javascript/bootstrap.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/javascript/select2/select2.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/javascript/spacewalk-essentials.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/javascript/spacewalk-checkall.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/rhn/dwr/engine.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/rhn/dwr/util.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/rhn/dwr/interface/DWRItemSelector.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/javascript/jquery.timepicker.js?cb=${cb_version}"></script>
+<script src="https://head-server.tf.local/javascript/bootstrap-datepicker.js?cb=${cb_version}"></script>
+<script src='https://head-server.tf.local/javascript/momentjs/moment-with-langs.min.js?cb=${cb_version}' type='text/javascript'></script>
```
Comment (cbosdo) [inline web/html/src/.storybook/preview-head.html]: This is really terrible to hardcode a server name... Couldn't we at least use the `https://localhost:8080` that is created by yarn proxy? But then this means that you need to run `yarn proxy --server https://head-server.tf.local` and then `yarn run storybook`.

At least this is working fine for me here.

File: web/html/src/components/dialog/dialog.stories.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { DeleteDialog } from "./DeleteDialog";
+import { ModalLink } from "./ModalLink";
+
+
+storiesOf('Dialogs', module)
+  .add('delete dialog', () => (
+    <div>
+      <ModalLink
```
Comment (cbosdo) [inline web/html/src/components/dialog/dialog.stories.js]: May be we want to have something more visible to click on

File: web/html/src/components/tree/tree.stories.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { Tree } from './tree';
+
+const simpleTreeData = {
+  rootId: "2",
+  items: [
+    {id: "2", children: ['3', '4', '8']},
+    {id: "3", data: {key: "three", description: "I am 3"}},
+    {id: "4", data: {key: "for", description: "I am 4 children"}, children: ["5", "6", "7"]},
```
Comment (cbosdo) [inline web/html/src/components/tree/tree.stories.js]: `s/for/four/`

The `I am 4 children` and the same for the following sounds a little weird. I'ld rather see something like `I am #4, with children`

File: web/html/src/.storybook/preview-head.html (hunk 1 above)
Comment (lneves12) [reply web/html/src/.storybook/preview-head.html]: Thanks, It's a good idea! At least we will not need to adapt that file in most of the times. If for some reason we don't want to run the proxy server before we can always adapt that file manually.

This still doesn't solve the problem if we want to deploy the styleguide somewhere, but that will be an effort for another time. To achieve that result we need to either migrate the legacy less->css building of the branding package in OBS to webpack or to somehow inline all the external dependencies inside preview.html. But neither of these 2 solutions will be fast to implement :)

File: web/html/src/components/dialog/dialog.stories.js (hunk 2 above)
Comment (lneves12) [reply web/html/src/components/dialog/dialog.stories.js]: Screenshot from 2019-08-30 11-51-21

I don't know how to do it without modifying the ModalLink component :P But I tried to center it inside a bigger container.  It's looking better now

**Comment by lneves12** [comment]
Thanks for the review @cbosdo 

All your comments were addressed :)

**Comment by lneves12** [comment]
You can rebase your PR on top of master now :)

I will update the frontend instructions on how to run it and backport it to 4.0.

**Comment by cbosdo** [comment]
Could you add some doc about it in https://github.com/uyuni-project/uyuni/wiki/Frontend-Development-Environment ?

**Comment by lneves12** [comment]
> Could you add some doc about it in https://github.com/uyuni-project/uyuni/wiki/Frontend-Development-Environment ?

That's exactly what I am doing atm ;)

**Comment by lneves12** [comment]
https://github.com/uyuni-project/uyuni/wiki/Frontend-Development-Environment#react-style-guide-of-reusable-components

**Pull request merged**